recipes/abseil/all/conanfile.py

```diff
@@ -0,0 +1,158 @@
+import os
+import glob
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+
+
+class ConanRecipe(ConanFile):
+    name = "abseil"
+
+    description = "Abseil Common Libraries (C++) from Google"
+    topics = ("algorithm", "container", "google", "common", "utility")
+
+    homepage = "https://github.com/abseil/abseil-cpp"
+    url = "https://github.com/conan-io/conan-center-index"
+
+    license = "Apache-2.0"
+
+    settings = "os", "arch", "compiler", "build_type"
+
+    options = {"fPIC": [True, False]}
+    default_options = {"fPIC": True}
+
+    generators = "cmake"
+    short_paths = True
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = glob.glob('abseil-cpp-*/')[0]
+        os.rename(extracted_dir, self._source_subfolder)
+        tools.replace_in_file(
+            os.path.join(self._source_subfolder, "CMakeLists.txt"),
+            "project(absl CXX)", """project(absl CXX)
+include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
+conan_basic_setup()""")
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        minimal_cpp_standard = "11"
+        if self.settings.compiler.cppstd:
+            tools.check_min_cppstd(self, minimal_cpp_standard)
+
+        minimal_version = {
+            "Visual Studio": "14",
+        }
+
+        compiler = str(self.settings.compiler)
+        if compiler not in minimal_version:
+            self.output.warn(
+                "%s recipe lacks information about the %s compiler support" % (self.name, compiler))
+            self.output.warn(
+                "%s requires a compiler that supports at least C++%s" % (self.name, minimal_cpp_standard))
+            return
+
+        version = tools.Version(self.settings.compiler.version)
+        if version < minimal_version[compiler]:
+            raise ConanInvalidConfiguration(
+                "%s requires at least %s %s" % (self.name, compiler, minimal_version[compiler]))
+
+    def _configure_cmake(self):
+        cmake = CMake(self)
+        cmake.definitions["BUILD_TESTING"] = False
+        cmake.configure(
+            source_folder=self._source_subfolder
+        )
+        return cmake
+
+    def build(self):
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def package(self):
+        self.copy("LICENSE", dst="licenses", src=self._source_subfolder)
+        cmake = self._configure_cmake()
+        cmake.install()
+        tools.rmdir(os.path.join(self.package_folder, "lib", "cmake"))
+
+    def package_info(self):
```

Comment:
I don't like it for all versions. From my experience, Abseil changes the library constantly. I would say you should create one recipe per version. However, creating one recipe version is boring, and only the cpp_info.libs will change. So, we could accept the lib list in conandata.yml, but it requires a filter there is conan-center hook.

Reply:
A good idea! The `all` folder is kinda a placeholder. Shouldn't be a problem to split recipes into different folders later on, isn't it?

Reply:
I not a big fan of the idea of having this information in the conandata.yml file, as it will be harder to read and it is mainly used (in this repo) for thing related to sources. But you can always make the libs conditional to the version of the recipe if you want to keep on recipe for all of them (easier to maintain. @Minimonium in case the recipe diverges too much we can always split it to one for each version